File: Easy/sqrt(x)/sqrt(x).cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
    public:
        int mySqrt(int x) {
            if(x == 0){
                return 0;
            }
            int left = 1, right = x, answer = 0;
            while(left <= right){
                int mid = left + (right - left) / 2;
                if(mid <= x / mid){
                    answer = mid;
                    left = mid + 1;
                }else{
                    right = mid - 1;
                }
            }
            return answer;
        }
};

//Example usage:
int main(){
    Solution solution;
    int x1 = 4;
    cout << "The integer square root of " << x1 << " is: " << solution.mySqrt(x1) << endl;

    int x2 = 8;
    cout << "The integer square root of " << x2 << " is: " << solution.mySqrt(x2) << endl;
}

File: Easy/sqrt(x)/sqrt(x).py
from typing import List

def mySqrt(x: int) -> int:
    if x < 2:
        return x
    left, right, answer = 1, x, 0
    while left <= right:
        mid = (left + right) // 2
        if(mid * mid) == x:
            return mid
        elif(mid * mid) < x:
            left = mid + 1
            answer = mid
        else:
            right = mid - 1
    return answer

if __name__ == "__main__":
    print("Testing mySqrt function:")
    print("For the input 4, the output is:", mySqrt(4)) #Output: 2
    print("For the input 8, the output is:", mySqrt(8)) #Output: 2
